### DISCUSSION TITLE
Torchscript transformer models

### DISCUSSION THREAD

**Discussion by jensprout** [explosion/spaCy#9037] [Help: Coding & Implementations] [answered]
Apologies for a newbie question... Is it possible to use a traced torchscript model (.pt file) for the transformer pipeline component? Using the Huggingface transformers library, I can load a traced model for inference via torch.jit.load("traced_bert.pt"). I'm not sure how to accomplish something similar with Spacy. 

For background context, I'm working toward deploying a model on AWS Inferentia EC2 instances. Spacy works fine out of the box on Inferentia, but to see any speedups with transformer models, you need to use AWS's Atom SDK to compile an optimized torchscript model. I'm trying to figure out how to get all the components to play nicely together.

**Comment by polm** [comment, marked as answer]
Sorry for the delayed reply to this. We haven't done that or heard of anyone doing it, so it's hard to say. 

I think in theory it should work but there might not be a way to specify the model file at the moment. Maybe poke around in the spacy-transformers implementation to see if you can figure out where it should go?

Sorry I can't be of more help, but thanks for making us aware of this usage pattern.

**Reply by jensprout** [reply]
Thanks! I'll give it a look! 

Even if you haven't had a ton of requests for jit support yet, I hope you'll consider adding it in the future to support higher-volume pipelines. Tests with traced models sped up my throughput by several orders of magnitude. I'm currently working with about 1.1M words per second, so any extra speedup helps a lot.

**Reply by jensprout** [reply]
Github isn't allowing me to mark this as the answer, but consider it done! Thank you!
